pkg/testcase/dualstack.go

```diff
@@ -127,14 +128,37 @@ func TestIPFamiliesDualStack(deleteWorkload bool) {
 }
 
 func TestIngressWithPodRestartAndNetPol(cluster *factory.Cluster, deleteWorkload bool) {
-	// Deploy server and client pods
-	err := shared.ManageWorkload("apply", "k3s_issue_10053_ns.yaml",
-		"k3s_issue_10053_pod1.yaml", "k3s_issue_10053_pod2.yaml")
-	Expect(err).NotTo(HaveOccurred(), "failed to deploy initial manifests")
+	serverPodIP, err := getPodIP(cluster)
+	Expect(err).NotTo(HaveOccurred(), "setupPods failed")
 
-	// Ensure the pods are running and retrieve the correct pod IP
-	var serverPodIP string
+	err = validatePodConnectivity(cluster, serverPodIP, deleteWorkload)
+	Expect(err).NotTo(HaveOccurred(), "validatePodConnectivity failed")
+}
```

Comment:
Rearranging this makes it a bit confusing to me what all is happening. I think it just comes down to function naming at the end of the day. Would it work/pass linting if we did this instead?

```go
func TestIngressWithPodRestartAndNetPol(cluster *factory.Cluster, deleteWorkload bool) {
	// Deploy server and client pods
	err := shared.ManageWorkload("apply", "k3s_issue_10053_ns.yaml",
		"k3s_issue_10053_pod1.yaml", "k3s_issue_10053_pod2.yaml")
	Expect(err).NotTo(HaveOccurred(), "failed to deploy initial manifests")
	filters := map[string]string{
		"namespace": "test-k3s-issue-10053",
	}
	serverPodIP := getServerPodIP(cluster, filters)

	// Ensure connectivity from client pod to server pod
	assert.ValidateIntraNSPodConnectivity("test-k3s-issue-10053", "client", serverPodIP, "Hostname: server")

	// Deploy network policy that explicitly allows access to the server pod
	err = shared.ManageWorkload("apply", "k3s_issue_10053_netpol.yaml")
	Expect(err).NotTo(HaveOccurred(), "whoami pod failed to deploy")

	// Ensure connectivity from client pod to server pod BEFORE restarting the server
	assert.ValidateIntraNSPodConnectivity("test-k3s-issue-10053", "client", serverPodIP, "Hostname: server")

	// Redeploy server pod and ensure it is up and running again. Retrieve its new IP.
	err = shared.ManageWorkload("delete", "k3s_issue_10053_pod1.yaml")
	Expect(err).NotTo(HaveOccurred(), "whoami pod failed to delete")
	err = shared.ManageWorkload("apply", "k3s_issue_10053_pod1.yaml")
	Expect(err).NotTo(HaveOccurred(), "whoami pod failed to redeploy")
	serverPodIP = getServerPodIP(cluster, filters)

	// Ensure connectivity from client pod to server pod AFTER restarting the server
	assert.ValidateIntraNSPodConnectivity("test-k3s-issue-10053", "client", serverPodIP, "Hostname: server")

	if deleteWorkload {
		err = shared.ManageWorkload("delete", "k3s_issue_10053_ns.yaml")
		Expect(err).NotTo(HaveOccurred())
	}
}

func getServerPodIP(cluster *factory.Cluster, filters map[string]string) string {
	var podIP string
	Eventually(func(g Gomega) {
		pods, poderr := shared.GetPodsFiltered(filters)
		g.Expect(poderr).NotTo(HaveOccurred())
		g.Expect(pods).NotTo(BeEmpty())

		for _, pod := range pods {
			processPodStatus(cluster, g, pod,
				assert.PodAssertRestart(),
				assert.PodAssertReady())

			if pod.Name == "server" {
				podIP = pod.IP
			}
		}
	}, "120s", "5s").Should(Succeed())
	return podIP
}
```

Reply:
will try

Reply:
it works

Reply:
@rancher-max

Reply:
https://github.com/rancher/distros-test-framework/pull/118/commits/ec4cac03d0ddebb2aa16174322490fb7401e8fe9